### ISSUE TITLE
mfs: unixfs 1.5 touch and chmod error on `raw` block

### ISSUE THREAD

**Issue by lidel** [ipfs/go-libipfs#660]
> Discussed with @gammazero  and decided to extract this to separate issue, to unblock testing https://github.com/ipfs/boxo/pull/653 in Kubo 0.30.0-rc1

After https://github.com/ipfs/boxo/pull/653 we now have a bug in `ipfs files touch` and `ipfs files chmod`. They fail when we execute them against `raw` blocks.

Repro steps with https://github.com/ipfs/kubo/pull/10478:

```console
$  echo -n "hello path gw" | ipfs add --cid-version 1 -Q  --raw-leaves --to-files /raw-test-1
bafkreiehuhfvxn7w4i5hexlhq4a3y7jf5kfogsvmm5zlfxg5poj4oqozyu

$ ipfs files touch /raw-test-1
Error: expected a ProtoNode as internal node

$ ipfs files chmod 0644 /raw-test-1
Error: expected a ProtoNode as internal node
```

Hopefully we can add code that detects when node is a raw block and upgrade it to dag-pb before setting mode/time ([somewhere around here](https://github.com/ipfs/boxo/blob/f35ddcbd4de7d9148fe6bea344260d6e3ea8be1d/mfs/file.go#L235-L252)). We have some prior art for detecting Raw blocks [here](https://github.com/ipfs/boxo/blob/88beadf0804607dedc7a4aa12cfef47381a74f89/mfs/file.go#L78-L98).

**Comment by gammazero** [comment]
Fixed by PR #653
